### ISSUE TITLE
using format_source on output collection elements not working

### ISSUE THREAD

**Issue by shiltemann** [galaxyproject/galaxy#2431]
I have the following in my tool xml:

``` xml
<inputs>
    <param name="infile" type="data" format="someformat" label="lab"/>
</inputs>
<outputs>
    <collection name="coll" type="list" label="lab">            
        <discover_datasets pattern=".*?\.(?P&lt;designation&gt;.*)\.suffix" format_source="infile"/>
    </collection>
</outputs>
```

but the format of the collection elements gets set to `None` instead of the desired `someformat`

i.e. in planemo I get:
`Dataset metadata verification for [file_ext] failed, expected [someformat] but found [None].`

same results if I put  `format_source` on the collection tag instead.

ping @jmchilton 

edit: on a related note, it would also be handy if `change_format` tags worked for collection elements

**Comment by nsoranzo** [comment]
By looking at `test/functional/tools/multi_output_configured.xml` it seems that adding `format="infile"` or `ext="infile"` to `<discover_datasets>` may work, but this still needs to be honed.

**Comment by shiltemann** [comment]
`format="someformat"`works fine there, but `format="infile"` just gets me

`Dataset metadata verification for [file_ext] failed, expected [someformat] but found [infile].`

this isn't a big deal, but would be nice to have for input dataset parameters that allow multiple input formats or for subclass datatypes etc :)

**Comment by nsoranzo** [comment]
Ok, sorry, it was worth a try!

**Comment by shiltemann** [comment]
yeah it definitely was, thank you, hadn't tried that yet, would have been a nice workaround, but alas :)

**Comment by jmchilton** [comment]
Mostly for my only record keeping since I have so many branches - but very initial WIP at https://github.com/jmchilton/galaxy/tree/format_source_discovery.

**Comment by bernt-matthias** [comment]
Would really be nice :)

**Comment by bernt-matthias** [comment]
On the way: https://github.com/galaxyproject/galaxy/pull/11754
